### PR TITLE
Update the CDN (again)

### PR DESCRIPTION
**Pull request by rob-maron**
No linked issue.

This CDN update lets us specify which topics are valid at the application level. A subscription to an invalid topic will have their connection closed.

### CODE REVIEW

File: crates/hotshot/src/traits/networking/push_cdn_network.rs [hunk 1]
```diff
@@ -155,13 +158,18 @@ pub struct PushCdnNetwork<TYPES: NodeType> {
...
 pub enum Topic {
     /// The global topic
     Global = 0,
     /// The DA topic
     DA = 1,
 }
 
+/// Implement the `TopicTrait` for our `Topic` enum. We need this to filter
+/// topics that are not implemented at the application level.
+impl TopicTrait for Topic {}
```
Comment (jparr721) [inline crates/hotshot/src/traits/networking/push_cdn_network.rs:171]: Does this actually implement anything?

**Review by jparr721 — APPROVED** [review]
Looks like ci-1 flaked. I re ran it, but it shouldn't be due to your changes.

File: crates/hotshot/src/traits/networking/push_cdn_network.rs (hunk 1 above)
Comment (rob-maron) [reply crates/hotshot/src/traits/networking/push_cdn_network.rs:171]: The trait is an alias that makes sure it implements `TryFrom<u8>`, `Into<u8>`, `Clone`, `PartialEq`, etc to enable conversions to and from a u8

Comment (rob-maron) [reply crates/hotshot/src/traits/networking/push_cdn_network.rs:171]: It also uses the default implementation for a function that prunes topics based on whether or not they are convertible to the application-level type

**Pull request merged**